signatures.md

```diff
@@ -52,59 +51,89 @@ Note: `Message` is used here as the object being signed, but this interface shou
+        //   sig - a series of bytes representing a signature  
+        //   err - a standard error message indicating any process issues
         // In:
-        //    m - a series of bytes representing a message to be signed
-        //    sk - a private key which cryptographically links `M` to `sig`
+        //   msg - a series of bytes representing a message to be signed
+  		  //   addr - the Filecoin `Address` of the signer
+        //   sk - a private key which cryptographically links `M` to `sig`
         //
-        Sign(m Message, sk PrivateKey) (sig SignatureBytes, err error)
-
-        // Verify validates the statement: only `M` could have generated `sig`
+        Sign(msg Message, addr Address, sk PrivateKey) (sig SignatureBytes, err error)
+        
+        // Aggregate generates a BLS signature aggregate from a set of BLS 
+        // signatures. Associated `Address` must be passed to this function
+  		  // so that the aggregate can be verified. This function will pair a `sig` 
+  			// with an `Address` based on index in `sigs` and `addresses`.
+        //
+        // Out:
+        // 	 sig - a series of bytes representing a single BLS signature
+        //   err - a standard error message indicating any process issues
+        // In:
+     		//   sigs - an array of SignatureBytes {sig_1, sig_2..., sig_n}
+    		//   addrs - an array of Address {address_sig_1, address_sig_2..., address_sig_n}
+  			//
+  			Aggregate(sigs SignatureBytes, addrs Addresses)(sig SignatureBytes, err error)
+  
+  			// Verify validates the statement: only `M` could have generated `sig`
         // given the validator has a message `m`, a signature `sig`, and a
         // public key `pk`.
+  			//
+  			// Verify is context dependant. If more than one Address or Signature is passed, 
+  			// then Verify will assume that SignatureBytes is a BLS signature aggregate. If 
+  			// only one signature is passed then Verify will receive signature type context 
+  			// from Address. 
         //
         // Out:
-        //    valid - a boolean value indicating the signature is valid
-        //    err - a standard error message indicating any process issues
+        //   valid - a boolean value indicating the signature is valid
+        //   err - a standard error message indicating any process issues
         // In:
-        //    m - a series of bytes representing the signed message
-        //    pk - the public key belonging to the signer `M`
-        //    sig - a series of bytes representing a signature usually `r`|`s`
+        //   msgs - an array of bytes representing signed messages
+        //   addrs - an array of Address {address_sig_1, address_sig_2..., address_sig_n}
+        //   sigs - an array of SignatureBytes {sig_1, sig_2..., sig_n}
         //
-        Verify(m Messgage, pk PublicKey, sig SignatureBytes) (valid bool, err error)
+        Verify(msgs Messgages, addrs Addresses, sigs SignatureBytes) (valid bool, err error)
```

Comment:
use arrays to denote multiple things, so `msgs []Message`

Reply:
Was trying to avoid go-lang syntax but that's probably silly at this point. Will change.